contracts/src/Gateway.sol

```diff
@@ -210,7 +210,11 @@ contract Gateway is IGateway, IInitializable {
     }
 
     function channelNoncesOf(ParaID paraID) external view returns (uint64, uint64) {
-        Channel storage ch = _ensureChannel(paraID);
+        Channel storage ch = CoreStorage.layout().channels[paraID];
+        // The agent will not exist at first and needs to be created.
+        if (ch.agent == address(0)) {
+            return (0, 0);
+        }
```

Comment:
With solidity, null values are implicitly converted to zero values, so can drop the null check
```suggestion
```